ui/main/src/app/business/services/system-notification.service.ts

```diff
@@ -0,0 +1,161 @@
+/* Copyright (c) 2023, Alliander (http://www.alliander.com)
+ * See AUTHORS.txt
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of the OperatorFabric project.
+ */
+
+import {Injectable, OnDestroy} from "@angular/core";
+import {LightCard, Severity} from "@ofModel/light-card.model";
+import {merge, Subject} from "rxjs";
+import {LightCardsFeedFilterService} from "./lightcards/lightcards-feed-filter.service";
+import {LightCardsStoreService} from "./lightcards/lightcards-store.service";
+import {ConfigService} from "./config.service";
+import {OpfabEventStreamService} from "./opfabEventStream.service";
+import {AlertMessageService} from "./alert-message.service";
+import {LogOption, OpfabLoggerService} from "./logs/opfab-logger.service";
+import {MessageLevel} from "@ofModel/message.model";
+import {Router} from "@angular/router";
+import {filter} from "rxjs/operators";
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SystemNotificationService implements OnDestroy {
+    private static RECENT_THRESHOLD = 400;
+
+    private notificationConfigBySeverity: Map<Severity, NotificationConfig>;
+    private notificationEnabled: Map<Severity, boolean>;
+
+    private incomingCardOrReminder = new Subject();
+    private ngUnsubscribe$ = new Subject<void>();
+    private lastSentCardId: string;
+
+    private isServiceActive = true;
+
+    constructor(
+        private lightCardsFeedFilterService: LightCardsFeedFilterService,
+        private lightCardsStoreService: LightCardsStoreService,
+        private configService: ConfigService,
+        private opfabEventStreamService: OpfabEventStreamService,
+        private alertMessageService: AlertMessageService,
+        private logger: OpfabLoggerService,
+        private router: Router
+    ) {
+        // use to have access from cypress to the current object for stubbing method sendNotification
+        if (window['Cypress']) window['systemNotificationService'] = this;
```

Comment:
We are not gonna test this service yet with cypress , so you can remove this line